### PR TITLE
Allow git to be cross-compiled

### PR DESCRIPTION
**Pull request by bgamari**
###### Motivation for this change

Allow the user to disable `git`'s dependencies on `perl`, making it possible to cross-compile it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/git/default.nix [hunk 1]
```diff
@@ -62,6 +62,11 @@ stdenv.mkDerivation {
   NIX_LDFLAGS = stdenv.lib.optionalString (!stdenv.cc.isClang) "-lgcc_s"
               + stdenv.lib.optionalString (stdenv.isFreeBSD) "-lthr";
 
+  configureFlags = stdenv.lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
+    "ac_cv_fread_reads_directories=yes"
+    "ac_cv_snprintf_returns_bogus=no"
```
[the remark this answers is not in the document]
Comment (bgamari) [reply pkgs/applications/version-management/git-and-tools/git/default.nix]: Indeed these won't always be correct; we are trying to predict what the results would be if the autoconf tests were run on the host. Of course, this is a losing battle so the best we can do is guess. If you know of any notable cases where these answers would be wrong then do let me know.

[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/applications/version-management/git-and-tools/git/default.nix]: Improving that will not cause native rebuilds so easy to do.

[the remark this answers is not in the document]
Comment (bgamari) [reply pkgs/applications/version-management/git-and-tools/git/default.nix]: I should say that I also find this sort of hack quite unsatisfactory. I've needed to do this quite a bit while getting things cross-compiling but sadly haven't yet come up with a better option.

**Comment by LnL7** [comment]
@Ericson2314 This should go to staging, it rebuilds almost +5000 packages.

**Comment by matthewbauer** [comment]
It's already in evaluation 1451298 so probably too late:

https://hydra.nixos.org/eval/1451298

**Comment by LnL7** [comment]
Well, it's broken so nothing actually gets cached until the next eval.

**Pull request merged**